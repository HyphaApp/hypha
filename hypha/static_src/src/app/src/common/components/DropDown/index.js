import * as React from "react";
import PropTypes from 'prop-types';

import HelperComponent from "@common/components/HelperComponent";


const DropDown = props => {
  return <div className="form__group ">
    <label htmlFor={props.name} className="form__question form__question--choice_field select" required="">
      <span>{props.label}</span>
      {props.required ? <span className="form__required"> *</span> : ""}
    </label>
    <HelperComponent {...props.helperProps} />
    
    <div className="form__item">
      <div className="form__select">
        <select
          name={props.name}
          id={props.name}
          onChange={e => props.onChange(props.name, e.currentTarget.value)}
        >
          {(props.choices || []).map(choice => {
            return <option key={choice[0]} value={choice[0]} selected={props.value && props.value == choice[0] }>{choice[1]}</option>
          })}
        </select>
      </div>
    </div>
  </div>
}
DropDown.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  required: PropTypes.bool,
  choices: PropTypes.array,
  helperProps: PropTypes.object,
  value: PropTypes.node
}

DropDown.displayName = 'DropDown';
export default DropDown;
