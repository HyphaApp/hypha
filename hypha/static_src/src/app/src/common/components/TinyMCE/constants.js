export const API_KEY = "fufz3rqdwl1zs7gkxw5wou2g7t00i0eijtyzvy3bgklon5zv";
