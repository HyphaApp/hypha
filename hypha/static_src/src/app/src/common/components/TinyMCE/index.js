import * as React from "react";

import { Editor } from '@tinymce/tinymce-react';
import HelperComponent from "@common/components/HelperComponent";
import PropTypes from 'prop-types';


const TinyMCE = props => {
  return <div className="form__group ">
    <label htmlFor={props.name} className="form__question form__question--choice_field select" required="">
      <span>{props.label} </span>
      {props.required ? <span className="form__required"> *</span> : ""}
    </label>
    <HelperComponent {...props.helperProps} />
    <Editor
      initialValue={props.value}
      init={{
        ...(props.mce_attrs),
        menubar: false
      }}
      onChange={e => props.onChange(props.name, e.level.content)}
      id={props.name}
    />
  </div>
}
TinyMCE.propTypes = {
  label: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.node,
  helperProps: PropTypes.object,
  name: PropTypes.string,
  mce_attrs: PropTypes.node
}

TinyMCE.displayName = 'TinyMCE';
export default TinyMCE;
