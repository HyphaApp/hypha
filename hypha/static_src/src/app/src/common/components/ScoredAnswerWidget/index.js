import * as React from "react";
import DropDown from "@common/components/DropDown";
import TinyMCE from "@common/components/TinyMCE";
import HelperComponent from "@common/components/HelperComponent";
import PropTypes from 'prop-types';



class ScoredAnswerWidget extends React.Component {

  onChange = index => (name, value) => {
    const updatedValue = this.props.value
    updatedValue[index] = value;
    this.props.onChange(this.props.name, updatedValue)
  }

  render() {
    return <div className="form__group ">

      <label htmlFor={this.props.name} className="form__question form__question--scored_answer_field scored_answer_widget">
        <span>{this.props.label}</span>
        {this.props.required ? <span className="form__required"> *</span> : ""}
      </label>
      <HelperComponent {...this.props.helperProps} />

      <div className="form__item">
        <TinyMCE
          name={this.props.name}
          value={this.props.value[0]}
          onChange={this.onChange(0)}
          id={this.props.name}
          init={this.props.widget[0].mce_attrs}
        />
     
        <div className="form__select">
          <DropDown
              name={this.props.name}
              value={this.props.value[1]}
              onChange={this.onChange(1)}
              id={this.props.name}
              choices={this.props.kwargs.fields[1].choices}
          />
        </div>

      </div>
    </div>
  }
}

ScoredAnswerWidget.propTypes = {
  value : PropTypes.node,
  onChange: PropTypes.func,
  name: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  helperProps: PropTypes.object,
  widget: PropTypes.array,
  kwargs:  PropTypes.object
}

ScoredAnswerWidget.displayName = 'ScoredAnswerWidget';
export default ScoredAnswerWidget;
