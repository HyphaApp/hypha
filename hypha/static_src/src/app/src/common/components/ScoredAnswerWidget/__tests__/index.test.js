import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import ScoredAnswerWidget from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });
// import  "../index.scss";
import TinyMCE from "@common/components/TinyMCE";
import DropDown from "@common/components/DropDown";


describe("Test scoredanswerwidget component", () => {
  const name = "test name";
  const label = "test label";
  const required = false;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = ["3",4];
  const widget = ["a", "b"]
  const kwargs = { fields : [{"id": 1}, {"choices" : [1,2]}]}
  const onChange = jest.fn();

  const subject = shallow(<ScoredAnswerWidget
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    widget={widget}
    kwargs={kwargs}
    onChange = {onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.find('.form__required').length).toBe(0);
    expect(subject.children().length).toEqual(3);
    expect(subject.find('span').first().text()).toBe(label);
    expect(subject.containsMatchingElement(<TinyMCE />)).toEqual(true);

    subject.find('TinyMCE').props().onChange();
    expect(onChange).toHaveBeenCalledWith(name, [undefined,4])
    expect(onChange).toHaveBeenCalled();
    subject.find('DropDown').last().props().onChange();
    expect(onChange).toHaveBeenCalledWith(name, [undefined,undefined])
    expect(onChange).toHaveBeenCalledTimes(2);
  });

  test('render a scoredAnswerWidget component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test scoredanswerwidget component with required", () => {
  const name = "test name";
  const label = "test label";
  const required = true;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = ["3",4];
  const widget = ["a", "b"]
  const kwargs = { fields : [{"id": 1}, {"choices" : [1,2]}]}
  const onChange = jest.fn();

  const subject = shallow(<ScoredAnswerWidget
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    widget={widget}
    kwargs={kwargs}
    onChange = {onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__required').length).toBe(1);
  });

  test('render a scoredAnswerWidget component with required', () => {
    expect(subject).toMatchSnapshot();
  });

});
