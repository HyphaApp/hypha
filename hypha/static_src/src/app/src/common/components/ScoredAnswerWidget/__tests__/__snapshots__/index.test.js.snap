// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test scoredanswerwidget component render a scoredAnswerWidget component 1`] = `
<div
  className="form__group "
>
  <label
    className="form__question form__question--scored_answer_field scored_answer_widget"
    htmlFor="test name"
  >
    <span>
      test label
    </span>
  </label>
  <HelperComponent />
  <div
    className="form__item"
  >
    <TinyMCE
      id="test name"
      name="test name"
      onChange={[Function]}
      value="3"
    />
    <div
      className="form__select"
    >
      <DropDown
        choices={
          Array [
            1,
            2,
          ]
        }
        id="test name"
        name="test name"
        onChange={[Function]}
        value={4}
      />
    </div>
  </div>
</div>
`;

exports[`Test scoredanswerwidget component with required render a scoredAnswerWidget component with required 1`] = `
<div
  className="form__group "
>
  <label
    className="form__question form__question--scored_answer_field scored_answer_widget"
    htmlFor="test name"
  >
    <span>
      test label
    </span>
    <span
      className="form__required"
    >
       *
    </span>
  </label>
  <HelperComponent />
  <div
    className="form__item"
  >
    <TinyMCE
      id="test name"
      name="test name"
      onChange={[Function]}
      value="3"
    />
    <div
      className="form__select"
    >
      <DropDown
        choices={
          Array [
            1,
            2,
          ]
        }
        id="test name"
        name="test name"
        onChange={[Function]}
        value={4}
      />
    </div>
  </div>
</div>
`;
